src/Compilers/Core/Portable/Symbols/ISymbol.cs

```diff
@@ -213,6 +213,14 @@ public interface ISymbol : IEquatable<ISymbol>
         /// <returns>A formatted string representation of the symbol.</returns>
         string ToDisplayString(SymbolDisplayFormat format = null);
 
+        /// <summary>
+        /// Converts the symbol to a string representation with given nullability.
+        /// </summary>
+        /// <param name="nullability">The nullability to use for the symbol.</param>
+        /// <param name="format">Format or null for the default.</param>
+        /// <returns>A formatted string representation of the symbol.</returns>
+        string ToDisplayString(Nullability nullability, SymbolDisplayFormat format = null);
```

Comment:
Should move this to a more restricted set of types:
ITypeSymbol
IDiscardSymbol
other symbols that have a top-level nullability

Reply:
This feels like this would be unnecessary if Nullability were simply tracked in the ITypeSymbol...